bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BookKeeperApiTest.java

```diff
@@ -0,0 +1,300 @@
+                newCreateLedgerOp()
+                    .withAckQuorumSize(1)
+                    .withWriteQuorumSize(2)
+                    .withEnsembleSize(3)
+                    .withDigestType(DigestType.MAC)
+                    .withCustomMetadata(customMetadata)
+                    .withPassword("password".getBytes(StandardCharsets.UTF_8))
+                    .execute());) {
+
+            byte[] data = "foo".getBytes(StandardCharsets.UTF_8);
+            writer.append(ByteBuffer.wrap(data)).get();
+            writer.append(ByteBuffer.wrap(data)).get();
+            writer.append(Unpooled.wrappedBuffer(data)).get();
+
+            writer.append(ByteBuffer.wrap(data)).get();
+            long expectedEntryId = writer.append(ByteBuffer.wrap(data)).get();
+            assertEquals(expectedEntryId, writer.getLastAddConfirmed());
+        }
+    }
+
+    @Test
+    public void testWriteAdvHandle() throws Exception {
+        Map<String, byte[]> customMetadata = new HashMap<>();
+        customMetadata.put("test", "test".getBytes(StandardCharsets.UTF_8));
+        try (WriteAdvHandle writer
+            = result(newCreateLedgerOp()
+                .withAckQuorumSize(1)
+                .withWriteQuorumSize(2)
+                .withEnsembleSize(3)
+                .withDigestType(DigestType.MAC)
+                .withCustomMetadata(customMetadata)
+                .withPassword("password".getBytes(StandardCharsets.UTF_8))
+                .makeAdv()
+                .execute());) {
+
+            long entryId = 0;
+            byte[] data = "foo".getBytes(StandardCharsets.UTF_8);
+            writer.write(entryId++, ByteBuffer.wrap(data)).get();
+            writer.write(entryId++, ByteBuffer.wrap(data)).get();
+            writer.write(entryId++, Unpooled.wrappedBuffer(data)).get(1, TimeUnit.MINUTES);
+
+            writer.write(entryId++, ByteBuffer.wrap(data)).get(1, TimeUnit.MINUTES);
+            long expectedEntryId = writer.write(entryId++, ByteBuffer.wrap(data)).get(1, TimeUnit.MINUTES);
+            assertEquals(expectedEntryId, writer.getLastAddConfirmed());
+        }
+    }
+
+    @Test
+    public void testWriteAdvHandleWithFixedLedgerId() throws Exception {
+        Map<String, byte[]> customMetadata = new HashMap<>();
+        customMetadata.put("test", "test".getBytes(StandardCharsets.UTF_8));
+        try (WriteAdvHandle writer
+            = result(newCreateLedgerOp()
+                .withAckQuorumSize(1)
+                .withWriteQuorumSize(2)
+                .withEnsembleSize(3)
+                .withDigestType(DigestType.MAC)
+                .withCustomMetadata(customMetadata)
+                .withPassword("password".getBytes(StandardCharsets.UTF_8))
+                .makeAdv()
+                .withLedgerId(1234)
+                .execute());) {
+
+            assertEquals(1234, writer.getId());
+
+            long entryId = 0;
+            byte[] data = "foo".getBytes(StandardCharsets.UTF_8);
```

Comment:
I don't see the difference between `testWriteAdvHandle` and `testWriteAdvHandleWithFixedLedgerId`. I image you need to verify the different behavior between without and with ledger id. but the test cases don't actually do this validation.